### PR TITLE
check content attribute option

### PR DESCRIPTION
**Pull request by itkg-alavieille**
https://github.com/open-orchestra/open-orchestra-model-interface/pull/50
https://github.com/open-orchestra/open-orchestra-model-bundle/pull/280

### CODE REVIEW

File: BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php [hunk 1]
```diff
@@ -79,24 +80,29 @@ protected function checkFieldType(FieldTypeInterface $data, $type, Form $form)
...
-        foreach ($data->getOptions() as $option) {
-            if (!in_array($option->getKey(), $keys)) {
-                $data->removeOption($option);
+            foreach ($data->getOptions() as $option) {
+                if (!in_array($option->getKey(), $keys)) {
+                    $data->removeOption($option);
+                }
             }
+        } else {
+            $data->setOptions(new ArrayCollection());
```
Comment (nicolasThal) [inline BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php]: this should not be done here!

The setOption method should not be called to but a empty arrayCollection inside.

**Comment by nicolasThal** [comment]
Can you also modify the testCase to explain your modification?

File: BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php (hunk 1 above)
Comment (itkg-alavieille) [inline BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php]: create a method in Field Type class for clean array ?

Comment (nicolasThal) [inline BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php]: What does the options array contains before you do this action?

Comment (itkg-alavieille) [inline BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php]: @nicolasThal , In edition, it contains the old options of the old content attribute.  in the case where the new content attribute has no option, we can clear array.

Comment (nicolasThal) [inline BackofficeBundle/EventSubscriber/FieldTypeTypeSubscriber.php]: All right. So to be more readable, i would suggest to add a clearOptions method in the interface and in the class.

This method would take care of the new ArrayCollection()

**Comment by itkg-ngilain** [comment]
:+1:

File: BackofficeBundle/Tests/EventSubscriber/FieldTypeTypeSubscriberTest.php [hunk 2]
```diff
@@ -160,4 +163,44 @@ public function testPreSetDataWithNoTypeSet()
...
+    /**
+     * Test preSetData when type has no option
+     */
+    public function testPreSetDataWithTypeNoOption()
+    {
+        Phake::when($this->fieldType)->getType()->thenReturn('hidden');
+
+        $this->subscriber->preSetData($this->event);
+
+        Phake::verify($this->fieldType, Phake::times(0))->addOption(Phake::anyParameters());
```
Comment (nicolasThal) [inline BackofficeBundle/Tests/EventSubscriber/FieldTypeTypeSubscriberTest.php]: I think that you could have put allthose verification in a single method

Comment (itkg-alavieille) [inline BackofficeBundle/Tests/EventSubscriber/FieldTypeTypeSubscriberTest.php]: the verification in testPreSetDataWithTypeNoOption method and testPreSubmitWithTypeNoOption method ?

Comment (nicolasThal) [inline BackofficeBundle/Tests/EventSubscriber/FieldTypeTypeSubscriberTest.php]: yes

Comment (nicolasThal) [inline BackofficeBundle/Tests/EventSubscriber/FieldTypeTypeSubscriberTest.php]: Did I missed something or you have suppressed a testCase?

**Comment by itkg-alavieille** [comment]
@nicolasThal

**Comment by nicolasThal** [comment]
:+1:

**Pull request merged**